### PR TITLE
docs: replace "New in" with "Since" in version annotations

### LINKED ISSUE #5166: 📝 Docs: Rephrase "New in ..." notices on versions to say "Since ..."
### Documentation Request Checklist

- [X] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/main/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md)
- [X] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [X] I want to provide a PR to resolve this

### Overview

The docs on mochajs.org include a lot of notices like https://mochajs.org/#-watch-ignore-filedirectoryglob:

> New in v7.0.0

`v7.0.0` came out 4 years ago. We're on 10.x now. I don't think we need to keep any _"New in"_ notices prior to 10.

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by TG199**
## PR Checklist

- [x] Addresses an existing open issue: fixes #5166 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Changes version annotations from "New in vX.Y.Z" to "Since vX.Y.Z".

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -928,13 +928,13 @@ To ensure your tests aren't leaving messes around, here are some ideas to get st
...
 
-> _New in v10.7.0_
+> _Since v10.7.0_
 
 If set to `true`, Mocha returns exit code `0` even if there are failed tests during run.
 
 ### `--fail-zero`
 
-> _New in v9.1.0_
+> _Since in v9.1.0_
```
Comment (JoshuaKGoldberg) [inline docs/index.md:937]: [Typo] Here and elsewhere:

```suggestion
> _Since v9.1.0_
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Mostly looks great to me, thanks for the first PR @TG199! A few small typos but you've got it correct otherwise.
 
Black cat using a nail file. Caption: "excellent"

**Comment by JoshuaKGoldberg** [comment]
> - [ x ]

By the way, it's `[x]`, not `[ x ]` (note the spaces). A nice little github.com feature: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists

File: docs/index.md (hunk 1 above)
Comment (TG199) [reply docs/index.md:937]: Thanks for approving my first PR , I'll fix the typos in another PR.

**Comment by TG199** [comment]
> > * [ x ]
> 
> By the way, it's `[x]`, not `[ x ]` (note the spaces). A nice little github.com feature: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists

Noted!

File: docs/index.md [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ Some assertion libraries will supply their own diffs, in which case Mocha's will
 
 Mocha's own diff output does not conform to any known standards, and is designed to be human-readable.
 
-> _New in v9.2.1_
+> _Since in v9.2.1_
```
Comment (JoshuaKGoldberg) [inline docs/index.md:1011]: @TG199 did you mean to get this one too? 🙂

Comment (TG199) [reply docs/index.md:1011]: yikes, missed it.

**Review by JoshuaKGoldberg — APPROVED** [review]
📝 thanks!

**Pull request merged**